pkg/internal/testing/addr/manager.go

```diff
@@ -28,33 +32,68 @@ import (
 const (
 	portReserveTime   = 1 * time.Minute
 	portConflictRetry = 100
+	portFilePrefix    = "port-"
 )
 
+var (
+	cacheDir string
+)
+
+func init() {
+	baseDir, err := os.UserCacheDir()
+	if err != nil {
+		baseDir = os.TempDir()
+	}
+	cacheDir = filepath.Join(baseDir, "kubebuilder-envtest")
+	_ = os.MkdirAll(cacheDir, 0750)
+}
+
 type portCache struct {
-	lock  sync.Mutex
-	ports map[int]time.Time
+	lock sync.Mutex
 }
 
-func (c *portCache) add(port int) bool {
+func (c *portCache) add(port int) (bool, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
```

Comment:
Do we still need the lock if we lock through the fs?

Reply:
I think so, mostly because of the WalkDir and the fact that envtest could be spun up multiple times within a single package

Reply:
But the whole point of this change is to make the locking work cross-binary so this must be threadsafe without a lock anyways or not?

Reply:
PTAL